app/src/main/java/com/github/sdpteam15/polyevents/user/User.kt

```diff
@@ -0,0 +1,135 @@
+            get() {
+                try {
+                    val currentUser: FirebaseUser =
+                        FirebaseAuth.getInstance().currentUser ?: return null
+                    if (currentUser.uid != lastCurrentUserUid)
+                        invoke(lastCurrentUserUid)?.removeCache()
+                    return invoke(FirebaseUserAdapter(currentUser))
+                } catch (e: ExceptionInInitializerError) {
+                    return null
+                } catch (e: NoClassDefFoundError) {
+                    return null
+                }
+            }
+    }
+
+    private var profileList: MutableList<ProfileInterface>? = null
+    private var currentProfileId: Int = 0
+    var database: DatabaseInterface = DatabaseObject.Singleton
+
+    private fun FromDatabase(): MutableList<ProfileInterface> {
+        if (profileList == null)
+            profileList = database.getListProfile(UID, this).toMutableList()
+        if ((profileList as MutableList<ProfileInterface>).size == 0) {
+            val profile = Profile(Name)
+            (profileList as MutableList<ProfileInterface>).add(profile)
+            database.addProfile(profile, UID, this)
+        }
+        return profileList as MutableList<ProfileInterface>
+    }
+
+    override val ProfileList: List<ProfileInterface>
+        get() = FromDatabase() as List<ProfileInterface>
+
+    override val Name: String
+        get() = FirebaseUser.displayName
+    override val UID: String
+        get() = FirebaseUser.uid
+    override val Email: String
+        get() = FirebaseUser.email
+
+    /**
+     * Current application user profile id
+     */
+    var CurrentProfileId: Int
+        get() = currentProfileId
+        set(value) {
+            if (value < 0 || value >= ProfileList.size)
+                throw IndexOutOfBoundsException("value must be between 0 and ${ProfileList.size}")
+            else currentProfileId = value
+        }
+
+    override var CurrentProfile: ProfileInterface
+        get() = ProfileList[currentProfileId]
+        set(value) {
+            var i = 0
+            for (v in ProfileList) {
+                if (v == value) {
+                    CurrentProfileId = i
+                    break
+                }
+                ++i
+            }
+        }
+
+    override fun removeCache() {
+        profileList = null
+    }
+
+    override fun newProfile(name: String) {
+        val profileList = FromDatabase()
+        val profile = Profile(name)
+        profileList.add(profile)
+        database.addProfile(profile, UID, this)
+    }
+
+    override fun removeProfile(profile: ProfileInterface): Boolean {
+        val profileList = FromDatabase()
+        if (!(profileList as MutableList<ProfileInterface>).contains(profile))
+            return false
+        if (CurrentProfile == profile)
+            currentProfileId = 0
+        profileList.remove(profile)
+        database.removeProfile(profile, UID, this)
```

Comment:
Is it really needed to pass the UID as argument to removeProfile since the current instance (this) is passed and therefore the UID should be accessible as "this.UID" in the removeProfile method ?

Reply:
yes, because admins can modify other user's profiles